### PR TITLE
add check for disconnected components

### PR DESCRIPTION
**Pull request by knaaptime**
This adds a check and raises a warning if the W matrix is not fully connected. We can use this to add check in cases like https://github.com/pysal/pysal/issues/1037#issuecomment-393549165 for algos like maxp that require a fully connected graph and can raise an error if `W.components >1`

### CODE REVIEW

**Comment by ljwolf** [comment]
now this needs to rebase & rerun

**Comment by ljwolf** [comment]
the error persists...

**Comment by ljwolf** [comment]
irrelevant testing errors were resolved by #74, so we're good to go with this if a test is written.

**Comment by knaaptime** [comment]
Great. I’ll finish this test I started

File: libpysal/weights/weights.py [hunk 1]
```diff
@@ -188,6 +198,8 @@ def __init__(self, neighbors, weights=None, id_order=None,
             else:
                 warnings.warn("There are %d disconnected observations" % ni)
```
Comment (ljwolf) [inline libpysal/weights/weights.py]: could we combine these warnings?

File: libpysal/weights/weights.py [hunk 2]
```diff
@@ -158,8 +167,9 @@ class W(object):
     """
 
     def __init__(self, neighbors, weights=None, id_order=None,
-        silent_island_warning=False, ids=None):
```
Comment (ljwolf) [inline libpysal/weights/weights.py]: and these flags?

Comment (ljwolf) [reply libpysal/weights/weights.py]: maybe call it `silent=True`.

**Pull request merged**